sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -2863,6 +2871,130 @@ pi_result cuda_piEnqueueMemUnmap(pi_queue command_queue, pi_mem memobj,
   return ret_err;
 }
 
+// USM
+pi_result cuda_piextUSMHostAlloc(void **result_ptr, pi_context context,
+                                 pi_usm_mem_properties *properties, size_t size,
+                                 pi_uint32 alignment) {
+  PI_CHECK_ERROR(cuMemAllocHost(result_ptr, size));
+  assert(*result_ptr % alignment == 0);
+  return PI_SUCCESS;
+}
+
+pi_result cuda_piextUSMDeviceAlloc(void **result_ptr, pi_context context,
+                                   pi_device device,
+                                   pi_usm_mem_properties *properties,
+                                   size_t size, pi_uint32 alignment) {
+  assert(device != nullptr);
+  ScopedContext active(context);
+  PI_CHECK_ERROR(cuMemAlloc((CUdeviceptr *)result_ptr, size));
```

Comment:
try-catch-block currently (sigh) required around `PI_CHECK_ERROR` to translate it into return codes.